src/DebtRequest.hs

```diff
@@ -118,13 +118,16 @@ mkValidator contractInfo@ContractInfo{..} dat borrowerTn ctx = validate
     isItToCollateral :: TxOut -> Bool
     isItToCollateral txo = txOutAddress txo == collateralSc
 
+    collateralAmount :: TxOut -> Integer
+    collateralAmount txo = assetClassValueOf (txOutValue txo) (collateral dat)
+
     containsRequiredCollateralAmount :: TxOut -> Bool
     containsRequiredCollateralAmount txo =
-      collateralAmnt dat <= assetClassValueOf (txOutValue txo) (collateral dat)
+      collateralAmnt dat <= collateralAmount txo
 
     containsNewDatum :: TxOut -> Bool
     containsNewDatum txo = case U.getUpperBound ctx of
-      Just ub -> findDatumHash' (expectedNewDatum ub) (U.info ctx) == txOutDatumHash txo
+      Just ub -> findDatumHash' (expectedNewDatum ub (collateralAmount txo)) (U.info ctx) == txOutDatumHash txo
```

Comment:
I'd introduce `updatedCollateral = collateralAmount txo` in a `where`-clause and use that.